scripts/postinstall.js

```diff
@@ -80,20 +104,40 @@ const main = async () => {
     }
     console.log(' > %s: Is either missing or outdated, downloading', editionId);
 
-    const result = await download(link(editionId));
-    result.pipe(tar.t()).on('entry', (entry) => {
-      if (entry.path.endsWith('.mmdb')) {
-        const dstFilename = path.join(downloadPath, path.basename(entry.path));
-        entry.pipe(fs.createWriteStream(dstFilename));
-        entry.on('end', () => {});
-      }
-    });
+    const response = await request(link(editionId));
+    const entryPromises = [];
+    await new Promise((resolve, reject) =>
+      response.body
+        .pipe(zlib.createGunzip())
+        .pipe(tar.t())
+        .on('entry', (entry) => {
+          if (entry.path.endsWith('.mmdb')) {
+            const dstFilename = path.join(
+              downloadPath,
+              path.basename(entry.path)
+            );
+            console.log(`writing ${dstFilename} ...`);
+            entryPromises.push(
+              new Promise((resolve, reject) => {
+                entry
+                  .pipe(fs.createWriteStream(dstFilename))
+                  .on('finish', resolve)
+                  .on('error', reject);
+              })
+            );
+          }
+        })
+        .on('end', resolve)
+        .on('error', reject)
+    );
+    await Promise.all(entryPromises);
   }
 };
 
 main()
   .then(() => {
     // success
+    process.exit(0);
```

Comment:
Is it dangling without explicit exit?

Reply:
Yes, I couldn't figure out why. (Maybe a stream is left open?) Explicit exit fixed it though